src/Nancy/Responses/Negotiation/DefaultResponseNegotiator.cs

```diff
@@ -357,15 +362,29 @@ protected virtual string CreateLinkHeader(Url requestUrl, IEnumerable<KeyValuePa
             var fileName = Path.GetFileNameWithoutExtension(requestUrl.Path);
             var baseUrl = string.Concat(requestUrl.BasePath, "/", fileName);
 
-            var links = linkProcessors
-                .Select(lp => string.Format("<{0}.{1}>; rel=\"alternate\"; type=\"{2}\"", baseUrl, lp.Key, lp.Value));
+            var result = string.Empty;
+
+            foreach (var linkProcessor in linkProcessors)
+            {
+                if (result.Length > 0)
+                {
+                    result += ", ";
+                }
+
+                result += string.Concat("<", baseUrl, ".", linkProcessor.Key, ">; rel=\"alternate\"; type=\"", linkProcessor.Value, "\"");

```

Comment:
I think we have to make this `StringBuilder`-based. Currently this will allocate a lot of strings.